src/utils/eventListeners.ts

```diff
@@ -0,0 +1,131 @@
+import { ClientType } from '@colony/colony-js';
+import { Log } from '@ethersproject/abstract-provider';
+import { Contract, utils } from 'ethers';
+
+import { addEvent } from '../eventQueue';
+import networkClient from '../networkClient';
+import { ContractEventsSignatures } from '../types';
+import {
+  addEventToQueue,
+  getClientAndProviderForELG as getClientAndProvider,
+  getEventFilter,
+  mapLogToContractEvent,
+} from './events';
+import { verbose } from './logger';
+/**
+ * Generator method for events listeners
+ *
+ * It basically does away with all the boilerplate of setting up topics, setting
+ * the event listener, parsing the received event
+ */
+export const eventListenerGenerator = async (
+  eventSignature: ContractEventsSignatures,
+  contractAddress: string,
+  clientType: ClientType = ClientType.NetworkClient,
+): Promise<void> => {
+  let { client, provider } = await getClientAndProvider(
+    clientType,
+    contractAddress,
+  );
+
+  const filter = getEventFilter(eventSignature, contractAddress, clientType);
```

Comment:
Have moved these to separate file to separate them from the helpers I have written for `eventListenerGenerator`